src/player.rs

```diff
@@ -198,7 +207,8 @@ impl PlayerInternal {
             if self.state.0.lock().unwrap().status == PlayStatus::kPlayStatusPlay {
                 match decoder.as_mut().unwrap().packets().next() {
                     Some(Ok(packet)) => {
-                        match stream.write(&packet.data) {
+                    				let buffer = packet.data.iter().map(|x| x/100*(self.state.0.lock().unwrap().volume/655) as i16).collect::<Vec<i16>>();
```

Comment:
I find x * volume / 0xFFFF clearer. You probably need to upcast x to a `i32` first to avoid overflow